libs/wingsdk/src/std/struct.ts

```diff
@@ -0,0 +1,32 @@
+import { T1 } from "./generics";
+import { Code, InflightClient } from "../core";
+import { Json } from "./json";
+
+/**
+ * Base class for all structs
+ * 
+ * @typeparam T1
+ */
+export class Struct {
+  /**
+   * @internal
+   */
+  public static _toInflightType(): Code {
+    return InflightClient.forType(__filename, this.name);
+  }
+
+  private constructor() { }
+
+  /**
+   * Converts a json to struct
+   * 
+   * This macro takes a json object and the struct definition file name then calls the validate method
+   * 
+   * @macro ((j, s) => {const errs = s.validate(j); if(errs.length !== 0) {throw new Error(`Unable to create struct: ${s.name} from json: ${JSON.stringify(j)}`)} return j;})($args$, $self$)
+   */
+  public static fromJson(json: Json): T1 {
```

Comment:
Id like to implement the `tryFromJson` in an immediate followup PR just because that is going to need to do some changes to macro system (hydrating optionals)

here is the issue: https://github.com/winglang/wing/issues/3653